modules/standard/IO.chpl

```diff
@@ -5754,7 +5775,17 @@ private inline
 proc _setIfPrimitive(ref lhs:?t, rhs:?t2, argi:int):syserr where t!=bool&&_isIoPrimitiveType(t)
 {
   try {
-    lhs = rhs:t;
+    if isAbstractEnumType(t) {
+      return ERANGE;
```

Comment:
I would think you could read an abstract enum from a string? So should this check be in an `isIntType(t2)` block?

Reply:
I think we don't actually yet have a test of a totally abstract enum and reading it with %s.

Reply:
Good catch, thanks!